mac/Keyman4MacIM/Keyman4MacIM/KMOSVersion.m

```diff
@@ -0,0 +1,51 @@
+//
+//  KMOSVersion.m
+//  Keyman
+//
+//  Created by Randy Boring on 2/13/19.
+//  Copyright © 2019 SIL International. All rights reserved.
+//
+
+#import "KMOSVersion.h"
+
+static CFStringRef sSystemVersionPath = CFSTR("/System/Library/CoreServices/SystemVersion.plist");
+
+@implementation KMOSVersion
++(u_int16_t)SystemVersion
+{
+    NSDictionary *systemVersionDict = [NSDictionary dictionaryWithContentsOfFile:(__bridge NSString*)sSystemVersionPath];
+    if (NULL ==systemVersionDict)
+    {
+        NSLog(@"could not read system version dictionary file: %@", sSystemVersionPath);
+        return 0;
+    }
+    NSString *systemVersionString = [systemVersionDict objectForKey:@"ProductVersion"];
+    if (NULL ==systemVersionString)
+    {
+        NSLog(@"could not retrieve system version from dictionary at %@ with %d keys", sSystemVersionPath, (int)systemVersionDict.count);
+        return 0;
+    }
+    NSArray *systemVersionStringParts = [systemVersionString componentsSeparatedByString:@"."];
+    if (systemVersionStringParts.count < 2)
+    {
+        NSLog(@"Too few parts to the system version (probably): %@", systemVersionString);
+        return [systemVersionString intValue]; // hopefully this will at least be the major OS version, e.g., 10 (all versino so far!)
```

Comment:
typo versino

```suggestion
        return [systemVersionString intValue]; // hopefully this will at least be the major OS version, e.g., 10 (all versions so far!)
```